EIPS/eip-X.md

```diff
@@ -0,0 +1,220 @@
+
+```solidity
+interface CrossChainReceiver {
+  function relayer() external view returns (address);
+}
+```
+
+```yaml
+- name: relayer
+  type: function
+  stateMutability: constant
+  outputs: 
+    - name: relayerAddress
+      type: address
+```
+
+**relayerChainId**
+
+Returns the id of the chain that the relayer lives on.
+
+```solidity
+interface CrossChainReceiver {
+  function relayerChainId() external view returns (uint256);
+}
+```
+
+```yaml
+- name: relayerChainId
+  type: function
+  stateMutability: constant
+  outputs: 
+    - name: relayerChainId
+      type: uint256
+```
+
+### Events
+
+**Executed**
+
+MUST be emitted when a relayed message is executed.
+
+```solidity
+interface CrossChainReceiver {
+  event Executed(
+      uint256 indexed nonce,
+      Call[] calls
+  );
+}
+```
+
+```yaml
+- name: Executed
+  type: event
+  inputs:
+    - name: nonce
+      indexed: true
+      type: uint256
+    - name: calls
+      type: Call[]
+```
+
+**RelayerSet**
+
+MUST be emitted when the relayer is initialized or updated.
+
+```solidity
+interface CrossChainReceiver {
+  event RelayerSet(
+      address indexed relayer,
+      uint256 indexed chainId
+  );
+}
+```
+
+```yaml
+- name: RelayerSet
+  type: event
+  inputs:
+    - name: relayer
+      indexed: true
+      type: address
+    - name: relayerChainId
+      indexed: true
+      type: uint256
+```
+
+# Rationale
+
+There are some notable design decisions worth talking about:
+
+- Relayers and receivers have a one-to-many relationship
+- Calls are relayed in batches
+- Receivers include both events and accessors to access the relayer
+
+The message receiver always authenticates the sender. This is the case whether contracts live on the same chain or across chains. That's why the relayer is unaware of the receiver, but the receiver is aware of the relayer. This approach is the most compatible with transport layers (namely Nomad!) and affords the ability to broadcast messages to multiple receivers.
+
+Calls are relayed in batches because it is such a common action. Rather than have implementors take different approaches to encoding multiple calls into the `data` portion, this spec includes call batching to take away any guess work.
+
+The receiver makes the relayer address and chainId available on-chain, and emits events when the Relayer is initialized or set. This makes it easy to trace cross-chain execution. Thanks to the events and nonce, an indexer can know exactly who the relayer and receiver were in the past and trace the execution of a call batch. With the on-chain accessor, external contracts can check to see who the current relayer is.
+
+# Backwards Compatibility
+
+This specification is compatible with existing governance systems as it offers simple cross-chain execution.
+
+# Implementation
+
+A basic implementation of the interfaces is available on [Github](https://github.com/pooltogether/cross-chain-relay-implementation).
+
```

Comment:
```suggestion
```
There is no Implementation section anymore.  There is Reference Implementation, but that section is optional and if included it should be inline code or CC0 licensed code at `../assets/eip-5164/*.md` and then relatively linked from here.